distributed/scheduler.py

```diff
@@ -7328,6 +7329,33 @@ async def get_call_stack(self, keys=None):
         response = {w: r for w, r in zip(workers, results) if r}
         return response
 
+    async def benchmark_hardware(self, comm=None):
+        out = defaultdict(lambda: defaultdict(list))
+
+        for mode in ["disk", "memory"]:
+            result = await self.broadcast(msg={"op": "benchmark_" + mode})
+            for d in result.values():
+                for size, duration in d.items():
+                    out[mode][size].append(duration)
+
+        workers = list(self.workers)
+        futures = []
+        for a, b in zip(workers[:-1], workers[1:]):
```

Comment:
my issue with this is that the first and last worker will have only a ping and a pong respectively, while every other worker will need to serve a ping to the next worker and a pong from the previous worker. If the network interfaces are half-duplex, this will skew measurements heavily.

An alternative algorithm that doesn't explode quadratically:
```python
workers = list(self.workers)
random.shuffle(workers)
for i in range(0, len(workers), 2):
    a, b = workers[i: i+1]
    self.rpc(a).benchmark_network(address=b)
```

Reply:
I'm less concerned, but I see no reason not to change to what you have here.  Added.